samples/net/lwm2m_client/src/lwm2m-client.c

```diff
@@ -390,7 +391,32 @@ void main(void)
 	client.tls_tag = TLS_TAG;
 #endif
 
+#if defined(CONFIG_HWINFO)
+	u8_t dev_id[16];
+	char dev_str[33];
+	ssize_t length;
+	int i;
+
+	(void)memset(dev_id, 0x0, sizeof(dev_id));
+
+	/* Obtain the device id */
+	length = hwinfo_get_device_id(dev_id, sizeof(dev_id));
```

Comment:
I wonder if it would be possible to provide dummy `hwinfo_get_device_id()` implementation if CONFIG_HWINFO=n in hwinfo.h. Because then we could use the `if (IS_ENABLED(CONFIG_HWINFO))` which is usually better to use than #ifdef's

Reply:
Sounds like a good idea to me but I'm not sure people are okay with the additional overhead. The default could be even be something like the string brought in by`CONFIG_BOARD` so if an implementation really cannot provide a real unique identifier it'll be at least a bit more useful than a batch of zeroes.

Reply:
> Sounds like a good idea to me but I'm not sure people are okay with the additional overhead

Hmm, what overhead do you mean here? If the CONFIG_HWINFO is not enabled, then having `if (IS_ENABLED(CONFIG_HWINFO))` would become `if (0)` which the compiler will optimize away. The benefit is that the code inside the "if" statement is compile checked in this case which is a good thing.

Reply:
Ah, okay, you want to place the check where the the `#ifdef` is now. Yeah, that'll work.